misk-jobqueue-testing/src/main/kotlin/misk/jobqueue/FakeJobQueue.kt

```diff
@@ -150,14 +182,30 @@ class FakeJobQueue @Inject constructor(
       }
     }
 
+    // Reenqueue deadlettered jobs outside of the main loop to prevent an infinite loop.
     result.forEach { job ->
       if (job.deadLettered) {
-        deadletteredJobs.getOrPut(queueName, ::ConcurrentLinkedDeque).add(job)
+        deadletteredJobs.getOrPut(job.queueName, ::ConcurrentLinkedDeque).add(job)
       }
     }
-
     return result
   }
+
+  private fun pollNextJob(jobs: Queue<FakeJob>, considerDelays: Boolean): FakeJob? {
+    if (!considerDelays) {
+      return jobs.poll()
+    }
+    val now = clock.instant()!!
+    while (true) {
+      val job = jobs.firstOrNull {
```

Comment:
firstOrNull will traverse the entire queue. I think you probably want to just `peek()` and check that

Reply:
You right, since queue is sorted by deliveryAt, using `peek()` will work here. I will fix that.
In my first version I reused `ConcurrentLinkedDeque` which was not sorted by deliveryAt, and later fixed to PriorityBlockingQueue.
I will also then make sure that dead letter queue never uses this branch (it is still on ConcurrentLinkedDeque)